### PR TITLE
chore: Upgrade to Elixir 1.15 and OTP 25

### PR DESCRIPTION
**Pull request by ryanatwork**
* Use Elixir 1.15 and OTP 25
* Upgrade ex_cldr to 2.37

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -9,13 +9,13 @@ defmodule Linguist.Mixfile do
...
 
-  @version "0.3.2"
+  @version "0.4.0"
 
   def project do
     [
       app: :linguist,
       version: @version,
-      elixir: "~> 1.6",
+      elixir: "~> 1.15",
```
Comment (barrieloydall) [inline mix.exs]: CI is still building on `1.10` which implies we still support back to that version, even with the `import` changes, so this should specify what is the min version we now support, and if we no longer support `1.6` should make that clear in the changelog

`ex_cldr:2.37.5` has `1.11`, so we should probably match that as the min we continue to support here in linguist

Comment (barrieloydall) [reply mix.exs]: `cldr_utils:2.24.2` also specifies `1.11` as the min version

Comment (ryanatwork) [reply mix.exs]: Ok updated to 1.11 as minimum version

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v0.4.0 (2024-01-31)
+* Upgrade to Elixir 1.15 and OTP 25
+* Upgrade `ex_cldr` to version 2.37
+
```
Comment (barrieloydall) [inline CHANGELOG.md]:
```suggestion
## v0.4.0 (2024-01-31)
* Add ci test and support Elixir 1.15 and OTP 25
* Upgrade `ex_cldr` to version 2.37
* Now requires Elixir ~> 1.11

```

**Review by barrieloydall — COMMENTED** [review]
LGTM - suggested update to changelog

**Review by barrieloydall — APPROVED** [review]

**Pull request merged**